adr/ADR 1: connaisseur-bootstrap-sentinel.md

```diff
@@ -0,0 +1,37 @@
+
+## Considered options
+
+### Option 1
+
+At the start of the Helm deployment, one can create a Pod named `connaisseur-bootstrap-sentinel` that will run for 30 seconds (default). Connaisseur Pods will report `Ready` if they can 1) access notary AND ( 2) the MutatingWebhookConfiguration exists OR 3) the `connaisseur-bootstrap-sentinel` Pod is still running ).
+
+### Option 2
+
+Let Connaisseur's Pod readiness stay non-indicative of Connaisseur functioning and advertise that someone running Connaisseur has to monitor the MutatingWebhookConfiguration in order to ensure proper working.
+
+### Option 3
+
+Deploy MutatingWebhookConfiguration through Helm when Connaisseur Pods are healthy instead of when ready. Require Pod started and working notary connection for health and require additionally the existence of the MutatingWebhookConfiguration for readiness.
+
+## Decision outcome
+
+We chose option 1 over option 2, because it was important to us that a brief glance at Connaisseur's Namespace allows one to judge whether it is running properly. Option 3 was not chosen as the readiness status of Pods can be easily seen from the Service, whereas the health status would require querying every single Pod individually. We deemed that to be a very ugly, non-kubernetes-y solution and hence decided against it.
+
+### Positive Consequences
+
+If the Connaisseur Pods report `Ready` during the `connaisseur-bootstrap-sentinel`'s runtime, the MutatingWebhookConfiguration will be deployed by Helm. Otherwise, the deployment will fail forever (since then neither the MutatingWebhookConfiguration will be deployed, nor will the `connaisseur-bootstrap-sentinel` Pod ever re-run). This means, we get consistent deployment failures after the inital waiting period if something did not work out. Additionally, if the MutatingWebhookConfiguration gets removed for whatever reason during operation, Connaisseur Pods will be failing, indicating their failed dependency. Hence, monitoring the Connaisseur Pods is sufficient to ensure their working.
+
+### Negative Consequences
+
+On the other hand, if an adversary can deploy a Pod named `connaisseur-bootstrap-sentinel` to Connaisseur's Namespace, the Connaisseur Pods will always show `Ready` regardless of the MutatingWebhookConfiguration. However, if an adversary can deploy to Connaisseur's Namespace, chances are Connaisseur can be compromised anyways. More importantly, if not a single Connaisseur Pod is successfully deployed or if the notary healthcheck fails during the initial period of 30 seconds, then the deployment will fail regardless of possible recovery at a later time. Another issue would be the `connaisseur-bootstrap-sentinel` Pod being left behind, however since it has a very limited use-case we can also clean it up during the deployment, so apart from the minimal additional complexity of the deployment this is a non-issue.
```

Comment:
`More importantly, if not a single Connaisseur Pod is successfully deployed or if the notary healthcheck fails during the initial period of 30 seconds, then the deployment will fail regardless of possible recovery at a later time.`

What are you trying to say here? That if the deployment fails, an adversery can't recover it? Is this supposed to be a positive or negative consequece?

Reply:
No, what I'm saying is that if notary is down for the first 30s of the deployment, the deployment will definetly fail, even if notary recovers afterwards

Reply:
ah ok now i get it